dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java

```diff
@@ -801,9 +803,35 @@ public void testGetSchema_GivenFailuresInRelationshipField_SchemaShouldStillGene
             assertNotNull(titleFieldDefinition);
         } finally {
             APILocator.getContentTypeAPI(APILocator.systemUser()).delete(contentType);
+            // restore normal RelationshipAPI for ContentAPIGraphQLTypesProvider
+            ContentAPIGraphQLTypesProvider.INSTANCE.setFieldGeneratorFactory(
+                    new GraphQLFieldGeneratorFactory());
         }
     }
 
+    @NotNull
+    private void setMockRelationshipAPI(Field relationshipField)
+            throws DotDataException, DotSecurityException {
+        RelationshipAPI relationshipAPI = Mockito.mock(RelationshipAPI.class);
+        Mockito.when(relationshipAPI.
+                getRelationshipFromField(relationshipField, APILocator.systemUser()))
+                .thenReturn(null);
+
+        RelationshipFieldGenerator relationshipFieldGenerator =
+                new RelationshipFieldGenerator();
+
+        relationshipFieldGenerator.setRelationshipAPI(relationshipAPI);
+
+        // lets create a mocked FieldGeneratorFactory
+        GraphQLFieldGeneratorFactory fieldGeneratorFactory =
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'fieldGeneratorFactory' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)